File: BackEnd/src/domain/subjectOffering.ts
import { GradeId, SchoolId, SubjectId, UserId } from './ids';

export class SubjectOffering {
  subjectOfferingId: string;
  subjectId: SubjectId;
  gradeId: GradeId;
  teacherId: UserId;
  schoolId: SchoolId;
  schoolYear: string;
}

File: Frontend/GP/Models/SubjectOffering.cs
using System.ComponentModel.DataAnnotations;

namespace GP.Models
{
    public class SubjectOffering : BaseEntity
    {
        [Key]
        public int SubjectOfferingId { get; set; }
        public int SubjectId { get; set; }
        public Subject Subject { get; set; }
        public int GradeId { get; set; }
        public Grade Grade { get; set; }
        public int TeacherId { get; set; }
        public Teacher Teacher { get; set; }
        public int SchoolId { get; set; }
        public School School { get; set; }
        public string SchoolYear { get; set; }
    }
}
